storage/disk/disk_test.go

```diff
@@ -167,63 +167,83 @@ func TestPolicies(t *testing.T) {
 	})
 }
 
-func TestTruncate(t *testing.T) {
+func TestTruncateAbsoluteStoragePath(t *testing.T) {
 	test.WithTempFS(map[string]string{}, func(dir string) {
-		ctx := context.Background()
-		s, err := New(ctx, logging.NewNoOpLogger(), nil, Options{Dir: dir, Partitions: nil})
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer s.Close(ctx)
+		runTruncateTest(t, dir)
+	})
+}
 
-		txn := storage.NewTransactionOrDie(ctx, s, storage.WriteParams)
+func TestTruncateRelativeStoragePath(t *testing.T) {
+	dir := "foobar"
```

Comment:
This is a bit messy, but I think it's better to test it this way -- potentially polluting some work dir -- than not at all.